sample-resource-plugin/src/integrationTest/java/org/opensearch/sample/AbstractSampleResourcePluginTests.java

```diff
@@ -0,0 +1,71 @@
+package org.opensearch.sample;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakScope;
+import org.junit.runner.RunWith;
+
+import org.opensearch.security.spi.resources.ResourceAccessScope;
+import org.opensearch.test.framework.TestSecurityConfig;
+
+import static org.opensearch.sample.utils.Constants.RESOURCE_INDEX_NAME;
+import static org.opensearch.sample.utils.Constants.SAMPLE_RESOURCE_PLUGIN_PREFIX;
+import static org.opensearch.security.dlic.rest.support.Utils.PLUGIN_RESOURCE_ROUTE_PREFIX;
+
+/**
+ * These tests run with security enabled
+ */
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class AbstractSampleResourcePluginTests {
```

Comment:
Are these tests just for a demonstration on how to write tests for resource plugins or are they actually contributing to testing the functionality in the security project?

Reply:
They are meant to be run with CI to ensure that the resource-sharing functionality works as intended.
https://github.com/DarshitChanpura/security/blob/6d5c80fdf847985061a60ed984413f3b9fa791a9/.github/workflows/ci.yml#L127

Reply:
I am not sure if it makes sense to have tests for functionality of the `security` plugin in the `sample-resource-plugin` module. IMHO, as the functionality is in `src/main`, the tests should be also in `src/test`. At least that would be intuitive for developers working on the code.

Reply:
Moving the tests from `sample-resource-plugin/src/integrationTest` to `src/integrationTest`. Reason: The code being tested is also located in `src/main`. Thus, the average developers looking for tests would look for them also in `src/integrationTest` or `src/test`. 

On the other hand, a sample is just a sample and should not contribute to the verification of the main components.

Reply:
This has been updated. Code being tested is now under sample-resource-plugin/src/main.